ansible_collections/arista/cvp/plugins/module_utils/device_tools.py

```diff
@@ -1302,35 +1349,83 @@ def refresh_fqdn(self, user_inventory: DeviceInventory):
         MODULE_LOGGER.info("Inventory to refresh is %s", str(user_inventory.devices))
         for device in user_inventory.devices:
             MODULE_LOGGER.debug("Found device %s to refresh data", str(device.info))
-            if device.system_mac is not None and self.__search_by == Api.device.SYSMAC:
-                fqdn = self.get_device_facts(device_lookup=device.system_mac)[
-                    Api.device.FQDN
-                ]
-                MODULE_LOGGER.debug(
-                    "Get fqdn %s for device %s", str(fqdn), str(device.system_mac)
-                )
-                device.fqdn = fqdn
-                user_result.append(device.info)
-            elif (
-                device.serial_number is not None
-                and self.__search_by == Api.device.SERIAL
-            ):
-                fqdn = self.get_device_facts(device_lookup=device.serial_number)[
-                    Api.device.FQDN
-                ]
-                MODULE_LOGGER.debug(
-                    "Get fqdn %s for device %s", str(fqdn), str(device.serial_number)
-                )
-                device.fqdn = fqdn
-                user_result.append(device.info)
-            else:
-                MODULE_LOGGER.debug("Skipping following device: %s", device.info)
+            for per_device in cvp_inventory:
+                if (
+                    device.system_mac is not None and
+                    self.__search_by == Api.device.SYSMAC and
+                    per_device[self.__search_by] == device.system_mac
+                ):
+                    fqdn = per_device[Api.device.FQDN]
+                    MODULE_LOGGER.debug(
+                        "Get fqdn %s for device %s", str(fqdn), str(device.system_mac))
+                    device.fqdn = fqdn
+                    user_result.append(device.info)
+                elif (
+                    device.serial_number is not None and
+                    self.__search_by == Api.device.SERIAL and
+                    per_device[self.__search_by] == device.serial_number
+                ):
+                    fqdn = per_device[Api.device.FQDN]
+                    MODULE_LOGGER.debug(
+                        "Get fqdn %s for device %s", str(fqdn), str(device.serial_number))
+                    device.fqdn = fqdn
+                    user_result.append(device.info)
+                else:
+                    MODULE_LOGGER.debug("Skipping following device: %s", device.info)
+
+        MODULE_LOGGER.warning("Update list is: %s", str(user_result))
+        return DeviceInventory(data=user_result)
+
+    def refresh_hostname(self, cvp_inventory: list, user_inventory: DeviceInventory):
```

Comment:
Why do you still need all these "refresh_xxx" when you have all the information already? It would be more efficient to have one function (`__refresh_user_inventory`) that updates all missing info on `user_inventory` from `cvp_inventory` instead of looping through the entire inventory several times.

Reply:
Makes sense. Will make that change.